File: game-server/src/main/java/org/dm/event/object/ObjectExamineEvent.java
package org.dm.event.object;

import com.dm.game.world.entity.mob.player.Player;
import com.dm.net.packet.out.SendMessage;
import org.dm.objectexamines.ObjectExamines;

/**
 * @author Jire
 */
public final class ObjectExamineEvent implements ObjectEvent {

    private final int objectId;

    public ObjectExamineEvent(int objectId) {
        this.objectId = objectId;
    }

    @Override
    public void handle(Player player) {
        String examine = ObjectExamines.map.get(objectId);
        if (examine == null) {
            return;
        }

        if (!"null".equals(examine)) {
            player.send(new SendMessage(examine));
        }
    }

    public int getObjectId() {
        return objectId;
    }

}

File: game-server/src/main/kotlin/org/jire/tarnishps/event/object/ObjectExamineEvent.kt
package org.jire.tarnishps.event.`object`

import com.dm.game.world.entity.mob.player.Player
import com.dm.net.packet.out.SendMessage
import org.jire.tarnishps.objectexamines.ObjectExamines

/**
 * @author Jire
 */
class ObjectExamineEvent(val objectId: Int) : ObjectEvent {

    override fun handle(player: Player) {
        val examine = ObjectExamines.map.get(objectId) ?: return
        if (examine != "null") {
            player.send(SendMessage(examine))
        }
    }

}